src/auto-trace.js

```diff
@@ -1,6 +1,32 @@
 import { wrapObjectWithError, createError } from './auto-trace.helper.js';
 let globalMiddlewares = [];
 
+/**
+ * Returns a method that will handle error processing and throwing. Pass this as your onError arg in RxJS subscriptions.
+ * This is meant to replace all the other catch and stacktrace methods. https://xkcd.com/927/
+ *
+ * RxJS Subscription example:
+ *   myObs.subscribe(onComplete, catchError())
+ *
+ * If a callback is passed then that will be called with the processed error instead of throwing the error.
+ * e.g.
+ *   myObs.subscribe(onComplete, catchError((error, throwError) => {
+ *     error.showToast = false
+ *     throwError(error)
+ *   }))
+ */
+export function catchError(callback, extraContext) {
+  return asyncStacktrace((err) => {
+    if (callback) {
+      callback(err, function throwErr(err) {
+        setTimeout(() => { throw err })
```

Comment:
is there a reason to throw inside a setTimeout? I think this will force the throw to happen outside of any micro task (like a promise) and into the main event loop...

Reply:
honestly no idea if the setTimeout is needed, I'm just mimicking what catchSyncStacktrace and catchAsyncStacktrace do